include/rmm/mr/device/detail/stream_ordered_memory_resource.hpp

```diff
@@ -203,7 +211,7 @@ class stream_ordered_memory_resource : public crtp<PoolResource>, public device_
 
     if (size <= 0) { return nullptr; }
 
-    lock_guard lock(mtx_);
+    // lock_guard lock(mtx_);
```

Comment:
question: why comment out the lock?

Reply:
That is because of a deadlock that would otherwise trigger when the allocation results in cudf-spilling. In this case, cudf-spilling will find another buffer to spill and de-allocate its memory, which also 
requires  the lock. 

I haven't given this too much thought, but I think this could be handled with a _reentrant_ lock.